File: 2.Binary-Search/binary-search.py
def binary_search(array, item):
    low = 0
    high = len(array) - 1

    while low <= high:
        mid = (low + high) // 2
        guess = array[mid]

        if guess < item:
            low = mid + 1
        elif guess > item:
            high = mid - 1
        else:
            return mid

    return None


my_array = [i for i in range(1, 101)]
print(binary_search(my_array, 99))

File: 1. Binary Search/binary-search.php
<?php
    function binary_search($array, $item) {
        $low = 0;
        $high = sizeof($array) - 1;

        while ($low <= $high) {
            $mid = floor(($low + $high) / 2);
            $guess = $array[$mid];
            
            if ($guess < $item) {
                $low = $mid++;
            } else if ($guess > $item) {
                $high = $mid--;
            } else {
                return $mid;
            }
        }

        return NULL;
    }

    $array = range(1, 100);
    echo binary_search($array, 99);

?>